src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -244,6 +253,37 @@ public async Task<bool> TryGetAsync(ArrayBuilder<DiagnosticData> list, Cancellat
 
                     return true;
                 }
+
+                static (bool includeSyntax, bool includeSemantic) ProcessDiagnosticKinds(
+                    DiagnosticAnalyzer analyzer,
+                    DiagnosticKinds diagnosticKinds,
+                    ref bool seenCompilerAnalyzer)
```

Comment:
The seenCompiler logic seems pretty weird to include in this helper, can it just be extracted out and done in the caller?

Reply:
Ah, so I misunderstood your previous suggestion. I agree with you that its better to avoid this ref param and instead having this method take in `bool isCompilerAnalyzer` parameter directly and have all the perf logic in the caller. I'll make that change.